docs/fsharp/index.md

```diff
@@ -14,45 +13,69 @@ ms.assetid: ea27fb37-dad1-4bd4-a3cc-4f5c70767ae9
 
 # F# Guide
 
-F# is a functional programming language which runs on .NET.  In addition to supporting functional programming constructs, it also has object programming capabilities.  This hybrid of functional programming with object-oriented capabilities makes F# a pragmatic language for accomplishing any task.
+F# is a functional programming language that runs on .NET. It also has full support for objects, letting you blend functional and object programming for pragmatic solutions to any problem.
 
-## If You're New to F# #
+```fsharp
+open System // Get access to functionality in System namespace.
 
-If you're new to F#, begin with the [Tour of F#](tour.md) to get an overview of the language and some of its programming concepts.  If you're using Visual Studio, the Tutorial project template contains the same content.
+// Function: takes a name and produces a greeting.
+let getGreeting name =
+    sprintf "Hello, %s! Isn't F# great?" name
 
-## If You're Experienced with F# #
+// Use the EntryPoint attribute to run the program.
+[<EntryPoint>]
+let main args =
+    args                     // Use F# pipe operators to send the args into some functions.
+    |> Array.map getGreeting // Turn each name into a friendly greeting.
+    |> Array.iter printfn    // Print them!
 
-If you know your way around F#, or want to learn more about a specific language construct, see the [Language Reference](language-reference/index.md).  It's a thorough guide of all F# language capabilities.
+    0
+```
 
-Additionally, the [F# Core Library Reference](https://msdn.microsoft.com/visualfsharpdocs/conceptual/fsharp-core-library-reference) is a great resource for learning about FSharp.Core, the core library which is a part of F#.
+F# is about productivity at its heart. The tooling support for F# is ubiquitous and full of advanced features.
 
-## The F# Software Foundation
+## Learning F# #
 
-Although Microsoft is the primary developer of the F# language and its tooling, F# is also backed by an independent foundation, the F# Software Foundation (FSSF).
+[Tour of F#](tour.md) gives an overview of major language features with lots of code samples. This is recommended if you are new to F# and want to get a feel for how the language works.
```

Comment:
Would the tour also be useful for new developers? If not, is there a new programmers introduction to F#?

Reply:
Perhaps, but most of the data I've seen about FP indicates that it's not something new programmers get into. Not until it's taught in beginner courses in university, at least.

Reply:
That makes sense. I think the important point is to identify the audience so they can self-select the right locationn.